### PR TITLE
Add discharge pass flag

### LINKED ISSUE #349: Discharge flag to assert statements one by one
During the discharge pass, because assertions are sent to the solver all at once, if something is very slow it makes it difficult to figure out what is causing the solver to slow. The discharge pass already contains the capability to send statements one by one instead (as this is what allows us to generate diagnostics) so we should have a flag the lets us use this behavior by default for debugging purposes.

### PR DESCRIPTION
**Pull request by UnsignedByte**
Closes #349

### CODE REVIEW

**Comment by UnsignedByte** [comment]
Pretty simple to implement

File: src/ir_passes/discharge.rs [hunk 1]
```diff
@@ -244,7 +244,7 @@ impl Discharge {
     }
 
     /// Check whether the proposition is valid.
-    /// Returns a set of assignments if the proposition is not valid.
+    /// Panics with a set of assignments if the proposition is not valid.
```
Comment (rachitnigam) [inline src/ir_passes/discharge.rs:247]: Sorry, why will this panic?

**Review by rachitnigam — APPROVED** [review]
One question about the comment update but otherwise LGTM. Merge at discretion if there is no functional change in the implementation

File: src/ir_passes/discharge.rs (hunk 1 above)
Comment (rachitnigam) [reply src/ir_passes/discharge.rs]: I've fixed the comment and merging!

**Pull request merged**